Author: Pino Toscano <toscano.pino@tiscali.it>
Description: Use modern realpath() to avoid PATH_MAX
Last-Update: 2021-10-03
Forwarded: https://sourceforge.net/p/astyle/bugs/549/
--- a/src/astyle_main.cpp
+++ b/src/astyle_main.cpp
@@ -1379,13 +1379,12 @@
 // Return the full path name or an empty string if failed.
 string ASConsole::getFullPathName(const string& relativePath) const
 {
-	// ignore realPath attribute warning
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunused-result"
-	char fullPath[PATH_MAX];
-	realpath(relativePath.c_str(), fullPath);
-	return fullPath;
-#pragma GCC diagnostic pop
+	char *fullPath = realpath(relativePath.c_str(), nullptr);
+	if (fullPath == nullptr)
+		return string();
+	const string p(fullPath);
+	free(fullPath);
+	return p;
 }
 
 /**
